src/widgets/PrescriptionCartRow.js

```diff
@@ -0,0 +1,85 @@
+/* eslint-disable react/forbid-prop-types */
+/**
+ * mSupply Mobile
+ * Sustainable Solutions (NZ) Ltd. 2019
+ */
+
+import React from 'react';
+import { View, StyleSheet } from 'react-native';
+import PropTypes from 'prop-types';
+
+import { CircleButton } from './CircleButton';
+import { DropdownRow } from './DropdownRow';
+import { DetailRow } from './DetailRow';
+import { StepperRow } from './StepperRow';
+import { CloseIcon } from './icons';
+import { Seperator } from './Seperator';
+
+/**
+ * Renders a row in the PrescriptionCart consisting of a `StepperRow`, `DropdownRow`,
+ * `DetailsRow` and a close button.
+ *
+ * Is passed all details needed as this is a simple layout and presentation component.
+ *
+ * @prop {Object} item
+ * @prop {Func}   onChangeQuantity
+ * @prop {Func}   onRemoveItem
+ * @prop {Array}  options
+ * @prop {Func}   onOptionSelection
+ */
+export const PrescriptionCartRow = ({
+  item,
+  onChangeQuantity,
+  onRemoveItem,
+  options,
+  onOptionSelection,
+}) => {
+  const { itemName, totalQuantity, itemCode, price } = item;
+
+  const itemDetails = [
+    { label: 'Code', text: itemCode },
+    { label: 'Price', text: price },
+  ];
+
+  return (
+    <View>
+      <View style={localStyles.flexRow}>
+        <View style={localStyles.largeFlexRow}>
+          <StepperRow text={itemName} quantity={totalQuantity} onChangeValue={onChangeQuantity} />
+          <DetailRow details={itemDetails} />
+          <View style={{ height: 10 }} />
+          <DropdownRow
+            currentOptionText="Some direction"
+            options={options}
+            onSelection={onOptionSelection}
+            dropdownTitle="Directions"
+          />
+        </View>
+        <View style={localStyles.flexOne} />
+        <CircleButton IconComponent={CloseIcon} onPress={onRemoveItem} />
+      </View>
+      <Seperator width={1} />
+    </View>
+  );
+};
+
+PrescriptionCartRow.propTypes = {
+  item: PropTypes.object.isRequired,
+  onChangeQuantity: PropTypes.func.isRequired,
+  onRemoveItem: PropTypes.func.isRequired,
+  options: PropTypes.array.isRequired,
+  onOptionSelection: PropTypes.func.isRequired,
+};
+
+const localStyles = StyleSheet.create({
+  flexRow: { flex: 1, flexDirection: 'row' },
+  largeFlexRow: { flex: 10 },
+  flexOne: { flex: 1 },
```

Comment:
again, globalStyles?

Reply:
you're right - but global styles or a component `<FlexView flex={x}> ... </FlexView>` ?

Reply:
oh, yeah - the `<FlexView/>` component. That's a much better idea

Reply:
cool - will make it a new issue as I've spread this all over the place 🤣